compiler/ast/ast_types.nim

```diff
@@ -651,7 +651,7 @@ type
     tfPacked
     tfHasStatic
     tfGenericTypeParam
-    tfImplicitTypeParam
+    tfImplicitTypeParam  ## generic or concept inferred from formal parameter
```

Comment:
```suggestion
    tfImplicitTypeParam ## the type parameter was lifted from the routine's
                        ## formal parameters
```
Implicit type parameters are not inferred -- they're lifted.

Reply:
can we mention `generic or concept` here?

Reply:
I don't see why. `generic` doesn't make sense in this case, and concepts aren't anymore special than other generic types.

Reply:
What do you think of the following?
```suggestion
    tfImplicitTypeParam ## the type parameter was lifted from a generic type
                        ## used by a formal parameter
```

Reply:
ok